### DISCUSSION TITLE
Install a python dependency from Artifactory

### DISCUSSION THREAD

**Discussion by marcellorego** [prefix-dev/pixi#2696] [Q&A] [answered]
Hello everyone. I am new to pixi, and maybe this question is already answered. 
I have already a Pypi package type already published to a private repository (Artifactory) and both .whl and .tag.gz are available for use in another project.
This project uses pixi and pyproject.toml file, and I would like to import above mentioned package into this project as a dependency.
As my repository is private, how to specify in pyproject.toml file or any other configuration file, the credentials for authentication?
I tried the approach documented at https://pixi.sh/dev/advanced/authentication/#pypi-authentication but no success.
The only way I could get access to whl file was using a basic authentication directly to repository url, which is not ideal, as it exposes credentials.
Could I get some guidance in this matter?
Thank you!

**Comment by tdejager** [comment, marked as answer]
Hey!

Something from authentication should work, a `.netrc` should always work when basic auth does. At least, that's what I'm thinking. There is another uv feature that would help, but we've not integrated it yet: https://docs.astral.sh/uv/configuration/indexes/#providing-credentials

Not sure if artifactory can make use of keyring somehow?

**Reply by marcellorego** [reply]
Hey Tim, thank you for your reply.  
Yes, indeed, `.netrc` file does the job. 
There was a configuration conflict in my machine, and to avoid that, I isolated pixi building process into a Dockerfile.

Solution was:
Have below tables added to `pyproject.toml` file.
```
[tool.pixi.pypi-options]
# list of additional urls to search for packages
extra-index-urls = ["https://artifactory.<company>.com/artifactory/api/pypi/<repository key>/simple"]

[tool.pixi.pypi-dependencies]
private_dependency = ">=x.y.z"
```
Create a `.netrc` file with authentication parameters.
Specify `NETRC` environment variable to point to such file path.
Then, when running `pixi install` it was able to authenticate to our repository and pull the artifact.
